app/src/main/java/com/eveningoutpost/dexdrip/ui/helpers/BitmapUtil.java

```diff
@@ -86,6 +56,84 @@ public static Bitmap getTiled(final Bitmap source, final int xSize, final int yS
         return tiledBitmap;
     }
 
+    public static Bitmap getBackgroundBitmap(final int xSize, final int ySize, final String background_file) {
+        Bitmap tiledBitmap = null;
+        try {
+            tiledBitmap = BitmapLoader.bitmapFromBundleCache(background_file);
+            if (tiledBitmap == null) {
+                android.util.Log.d("NumberWall", "Regenerating image");
+
+                final Uri background_uri = Uri.parse(background_file);
+
+                Bitmap image_bitmap = null;
+                if (background_file.startsWith("content://")) {
+                    FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor();
+                    image_bitmap = BitmapFactory.decodeFileDescriptor(fileDescriptor);
+                } else {
+                    File file = new File(background_file);
+                    FileInputStream in = new FileInputStream(file);
+                    image_bitmap = BitmapFactory.decodeStream(in);
+                    in.close();
+                }
+
+                Bitmap rotated_bitmap = image_bitmap;
+                final Matrix image_matrix = new Matrix();
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
+                    ExifInterface exif = null;
+                    if (background_file.startsWith("content://")) {
+                        FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor(); // reset
+                        exif = new ExifInterface(fileDescriptor);
+                    } else {
+                        File file = new File(background_file);
+                        FileInputStream in = new FileInputStream(file);
+                        exif = new ExifInterface(in);
+                        in.close();
+                    }
+                    int rotation = exifOrientationToDegrees(exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL));
```

Comment:
`exif` could still be `null` here.

Reply:
actually, if I add an `if (exif == null)` there, Android Studio shows a warning that it's not necessary. I think that because on the `else` case it will throw an exception if it fails to open the file, otherwise it will not be null